openmc/summary.py

```diff
@@ -0,0 +1,229 @@
+            self._fast_materials[material.id] = material
+
+    def _read_surfaces(self):
+        for group in self._f['geometry/surfaces'].values():
+            surface = openmc.Surface.from_hdf5(group)
+            self._fast_surfaces[surface.id] = surface
+
+    def _read_cells(self):
+
+        # Initialize dictionary for each Cell's fill
+        cell_fills = {}
+
+        for key, group in self._f['geometry/cells'].items():
+            cell_id = int(key.lstrip('cell '))
+            name = group['name'][()].decode() if 'name' in group else ''
+            fill_type = group['fill_type'][()].decode()
+
+            if fill_type == 'material':
+                fill = group['material'][()]
+            elif fill_type == 'universe':
+                fill = group['fill'][()]
+            else:
+                fill = group['lattice'][()]
+
+            region = group['region'][()].decode() if 'region' in group else ''
+
+            # Create this Cell
+            cell = openmc.Cell(cell_id=cell_id, name=name)
+
+            if fill_type == 'universe':
+                if 'translation' in group:
+                    translation = group['translation'][()]
+                    translation = np.asarray(translation, dtype=np.float64)
+                    cell.translation = translation
+
+                if 'rotation' in group:
+                    rotation = group['rotation'][()]
+                    rotation = np.asarray(rotation, dtype=np.int)
+                    cell._rotation = rotation
+
+            elif fill_type == 'material':
+                cell.temperature = group['temperature'][()]
+
+            # Store Cell fill information for after Universe/Lattice creation
+            cell_fills[cell.id] = (fill_type, fill)
+
+            # Generate Region object given infix expression
+            if region:
+                cell.region = Region.from_expression(region, self._fast_surfaces)
+
+            # Add the Cell to the global dictionary of all Cells
+            self._fast_cells[cell.id] = cell
+
+        return cell_fills
+
+    def _read_universes(self):
+        for group in self._f['geometry/universes'].values():
+            universe = openmc.Universe.from_hdf5(group, self._fast_cells)
+            self._fast_universes[universe.id] = universe
+
+    def _read_lattices(self):
+        for group in self._f['geometry/lattices'].values():
+            lattice = openmc.Lattice.from_hdf5(group, self._fast_universes)
+            self._fast_lattices[lattice.id] = lattice
+
+    def _finalize_geometry(self, cell_fills):
+
+        # Keep track of universes that are used as fills. That way, we can
+        # determine which universe is NOT used as a fill (and hence is the root
+        # universe)
+        fill_univ_ids = set()
+
+        # Iterate over all Cells and add fill Materials, Universes and Lattices
+        for cell_id, (fill_type, fill_id) in cell_fills.items():
```

Comment:
The cell_fills you have defined in `_read_cells` method sets value as `(fill_type, fill)`, whereas you read those as `(fill_type, fill_id)`. If they represent same objects, you should use same representation.